mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/PrepareGameScreenState.kt

```diff
@@ -1,23 +1,31 @@
 package ch.epfl.sdp.mobile.ui.prepare_game
 
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.chess.ChessFacade
+import ch.epfl.sdp.mobile.application.chess.Match
+import ch.epfl.sdp.mobile.ui.social.Person
+import kotlinx.coroutines.CoroutineScope
+
 /**
  * State interface of the [PrepareGameScreen]
  * @property colorChoice chosen color
- * @property gameType chosen game type
+ * @property selectedOpponent the selected opponent whom to create a game with
  */
-interface PrepareGameScreenState {
-  var colorChoice: ColorChoice
-  var gameType: GameType
-}
+interface PrepareGameScreenState<P : Person> {
 
-/** Color choices for a chess game */
-enum class ColorChoice {
-  White,
-  Black
-}
+  /** Color choices for a chess game */
+  enum class ColorChoice {
+    White,
+    Black
+  }
 
-/** Game types for local and online chess games */
-enum class GameType {
-  Local,
-  Online
+  var colorChoice: ColorChoice
+  val opponents: List<P>
+  var selectedOpponent: P?
+  val user: AuthenticatedUser
+  val navigateToGame: (Match) -> Unit
+  val onPlayClick: (P) -> Unit
```

Comment:
I don't know if the view should be responsible to send the player to play with; to me, it sounds like something the ViewModel should be handling.

What would you think of having the following signatures instead ?

```kotlin
val selectedOpponent: P?
fun onOpponentClick(opponent: P)
fun onPlayClick()
```

where the `PrepareGameScreenState` implementation takes care of managing the selection state / swapping the opponents by itself ?

Reply:
Fixed in [5fb0802](https://github.com/epfl-SDP/android/pull/200/commits/5fb08021c954bdad0ad44de1e9e111b7a5eff163)